code/modules/mafia/roles.dm

```diff
@@ -256,3 +270,88 @@
 			to_chat(body,"<span class='danger'>Your attempt at killing [current_victim.body] was prevented!</span>")
 	current_victim = null
 
+//just helps read better
+#define FUGITIVE_NOT_PRESERVING 0//will not become night immune tonight
+#define FUGITIVE_WILL_PRESERVE 1 //will become night immune tonight
+
+/datum/mafia_role/fugitive
+	name = "Fugitive"
+	desc = "You're on the run. You can become immune to night kills exactly twice, and you win by surviving to the end of the game with anyone."
+	team = MAFIA_TEAM_SOLO
+	actions = list("Self Preservation")
+	var/charges = 2
+	var/protection_status = FUGITIVE_NOT_PRESERVING
+
+/datum/mafia_role/fugitive/New(datum/mafia_controller/game)
+	. = ..()
+	RegisterSignal(game,COMSIG_MAFIA_NIGHT_START,.proc/night_start)
+	RegisterSignal(game,COMSIG_MAFIA_NIGHT_END,.proc/night_end)
+	RegisterSignal(game,COMSIG_MAFIA_GAME_END,.proc/survived)
+
+/datum/mafia_role/fugitive/handle_action(datum/mafia_controller/game, action, datum/mafia_role/target)
+	. = ..()
+	if(!charges)
+		to_chat(body,"<span class='danger'>You're out of supplies and cannot protect yourself tonight.</span>")
+		return
+	if(protection_status == FUGITIVE_WILL_PRESERVE)
+		to_chat(body,"<span class='danger'>You decide to not prepare tonight.</span>")
+	else
+		to_chat(body,"<span class='danger'>You decide to prepare for a horrible night.</span>")
+	protection_status = !protection_status
+
+/datum/mafia_role/fugitive/proc/night_start(datum/mafia_controller/game)
+	if(protection_status == FUGITIVE_WILL_PRESERVE)
+		to_chat(body,"<span class='danger'>Your preparations are complete. Nothing could kill you tonight!</span>")
+		RegisterSignal(src,COMSIG_MAFIA_ON_KILL,.proc/prevent_death)
+
+/datum/mafia_role/fugitive/proc/night_end(datum/mafia_controller/game)
+	if(protection_status == FUGITIVE_WILL_PRESERVE)
+		charges--
+		UnregisterSignal(src,COMSIG_MAFIA_ON_KILL)
+		to_chat(body,"<span class='danger'>You are no longer protected. You have [charges] use[charges == 1 ? "" : "s"] left of your power.</span>")
+		protection_status = FUGITIVE_NOT_PRESERVING
+
+/datum/mafia_role/fugitive/proc/prevent_death(datum/mafia_controller/game)
+	to_chat(body,"<span class='userdanger'>You were attacked! Luckily, you were ready for this!</span>")
+	return MAFIA_PREVENT_KILL
+
+/datum/mafia_role/fugitive/proc/survived(datum/mafia_controller/game)
+	game.send_message("<span class='big yellow'>!! FUGITIVE VICTORY !!</span>")
+
+#undef FUGITIVE_NOT_PRESERVING
+#undef FUGITIVE_WILL_PRESERVE
+#undef FUGITIVE_NOW_PRESERVING
+
+/datum/mafia_role/obsessed
+	name = "Obsessed"
+	desc = "You're completely lost in your own mind. You win by lynching your obsession before you get killed in this mess."
+	team = MAFIA_TEAM_SOLO
+
+	var/datum/mafia_role/obsession
+	var/lynched_target = FALSE
+
+/datum/mafia_role/obsessed/New(datum/mafia_controller/game) //note: obsession is always a townie
```

Comment:
i worry about actually having everyone in the game when this happens. i'll move this to the end of day one to pick a townie to obsess over